landscape/lib/network.py

```diff
@@ -35,32 +35,16 @@ def is_up(flags):
     return flags & 1
 
 
-def get_active_interfaces():
-    """Generator yields (active network interface name, address data) tuples.
-
-    Address data is formatted exactly like L{netifaces.ifaddresses}, e.g.::
-
-        ('eth0', {
-            AF_LINK: [
-                {'addr': '...', 'broadcast': '...'}, ],
-            AF_INET: [
-                {'addr': '...', 'broadcast': '...', 'netmask': '...'},
-                {'addr': '...', 'broadcast': '...', 'netmask': '...'},
-                ...],
-            AF_INET6: [
-                {'addr': '...', 'netmask': '...'},
-                {'addr': '...', 'netmask': '...'},
-                ...], })
-
-    Interfaces with no IP address are ignored.
+def is_active(ifaddresses):
+    """Checks if interface address data has an IP address
+
+    @param ifaddresses: a dict as returned by L{netifaces.ifaddresses}
     """
-    for interface in netifaces.interfaces():
-        ifaddresses = netifaces.ifaddresses(interface)
-        # Skip interfaces with no IPv4 or IPv6 addresses.
-        inet_addr = ifaddresses.get(netifaces.AF_INET, [{}])[0].get('addr')
-        inet6_addr = ifaddresses.get(netifaces.AF_INET6, [{}])[0].get('addr')
-        if inet_addr or inet6_addr:
-            yield interface, ifaddresses
+    inet_addr = ifaddresses.get(netifaces.AF_INET, [{}])[0].get('addr')
+    inet6_addr = ifaddresses.get(netifaces.AF_INET6, [{}])[0].get('addr')
+    if inet_addr or inet6_addr:
```

Comment:
needless branching…
`return inet_addr or inet6_addr`

Reply:
I feel like I should leave this in due to the explicit return value, otherwise it would be a string or None instead of True/False